client/src/components/StreamField/blocks/StructBlock.js

```diff
@@ -96,9 +101,28 @@ export class StructBlock {
     }
     const error = errorList[0];
 
-    for (const blockName in error.blockErrors) {
-      if (hasOwn(error.blockErrors, blockName)) {
-        this.childBlocks[blockName].setError(error.blockErrors[blockName]);
+    // Non block errors
+    const container = this.container[0];
+    container
+      .querySelectorAll(':scope > .help-block.help-critical')
```

Comment:
I think ideally we use data attributes to select elements in JS, as in this case it also helps prevent accidentally removing elements that we do not create (e.g. if the developer for some reason has a custom panel with these classes on one of the inner elements). However, considering we also do this in `StreamBlock.js`, happy to leave this and refactor separately as a cleanup.

Reply:
I've avoided making any changes to the HTML rendering in these PRs (in fact I was pleasantly surprised that these changes _didn't_ require any updates to the jest snapshots :-) ) but I'm in favour of dealing with this in a follow-up PR. In #10131 this repeated code is moved into a shared include, which should help with that.